napari/_qt/qt_main_window.py

```diff
@@ -723,8 +723,6 @@ def _update_widget_states(self, e=None):
 
     def _reset_preference_states(self):
```

Comment:
I tried to see if I could determine a problem here with this function being called when the preferences dialog was opened and closed, and its not being called then for me.  

It *is* called when I reset defaults from the preferences dialog.  If you do not want to reset the blocked list in the plugin manager when defaults are restored, then we don't need this line.   I could see that being confusing since enabling and disabling a plugin happens (or it did happen) on the plugin install dialog.  If a user is resetting defaults, they may not think it will reset all their enabled/disabled plugin states (when that option is back).

I think this change makes sense; however, I couldn't replicate the problem this was trying to fix in the first place.

Reply:
> If you do not want to reset the blocked list in the plugin manager when defaults are restored, then we don't need this line.

This function is getting called even if you just hit the close button or the cancel button (not the restore default button).  So if restoring defaults is also depending on this line, that needs to be fixed elsewhere

Reply:
ah, I see now.  I was always clicking "ok" and it doesn't trigger, but I see it now with cancel or close.

Reply:
yep, see also: https://github.com/napari/napari/issues/3144 
I think preferences "sticking" is the standard behavior when closing a preferences window (at least on mac), so almost every time I open the window I close it with cmd-w (muscle memory)... and each time my blocked plugins get cleared 😅